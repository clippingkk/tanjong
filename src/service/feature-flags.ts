export const featureFlags = {
  enableSignUp: false
};