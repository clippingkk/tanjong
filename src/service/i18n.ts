import i18n, { Module, NewableModule } from 'i18next';
import { initReactI18next } from 'react-i18next';
import { I18nManager, NativeModules, Platform } from 'react-native'
import * as RNLocalize from "react-native-localize"

import en from '../l10n/en.json'
import zh from '../l10n/zhCN.json'
import ko from '../l10n/ko.json'
import { storage } from '../utils/storage';

const languageDetector: any = {
  type: 'languageDetector',
  async: true,
  detect: async (cb: any) => {
    const storedLng = storage.getString('lng')
    if (storedLng) {
      return Promise.resolve(storedLng)
    }
    const locale =
      Platform.OS === 'ios' ?
        NativeModules.SettingsManager.getConstants().settings.AppleLocale :
        NativeModules.I18nManager.localeIdentifier;

    const l = locale.split('_')[0]
    return Promise.resolve(l)
  },
  cacheUserLanguage: async (lng: string) => {
    storage.set('lng', lng)
  }
}

i18n
  .use(languageDetector)
  .use(initReactI18next)
  .init<any>({
    compatibilityJSON: 'v4',
    fallbackLng: 'en',
    resources: {
      en: {
        translation: en
      },
      zh: {
        translation: zh
      },
      ko: {
        translation: ko
      },
    },
    ns: ['translation'],
    defaultNS: 'translation',
    interpolation: {
      escapeValue: false // not needed for react
    },
    react: {
      useSuspense: false
    }
  });

export default i18n;