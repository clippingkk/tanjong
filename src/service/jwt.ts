export interface JwtPayload {
  id: string;
  exp: number;
  iat: number;
  iss: string;
}