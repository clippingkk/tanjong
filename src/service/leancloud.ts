export const LEANCLOUD = {
  APP_ID: 'aA4dKCxL9noDX5CyYp2AdnHl-9Nh9j0Va',
  APP_KEY: 'QQ7h7pbOY4xFWEap8fVuFtcl',
  SERVER_URL: 'https://clippingkk-lc.annatarhe.cn'
}
