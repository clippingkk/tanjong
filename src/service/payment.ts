import { request } from "../utils/apollo";

export type StripeError = {
  code: string;
  doc_url: string;
  status: number;
  message: string;
  param: string;
  request_id: string;
  request_log_url: string;
  type: string;
}

export function getPaymentSheet() {
  return request<{
    paymentIntent: string
    ephemeralKey: string
    customer: string
  }>('/v2/payment-sheet', {
    method: 'POST'
  })
}
