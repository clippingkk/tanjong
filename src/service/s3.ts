import { API_HOST } from "../constants/config"
import { getLocalToken } from "../utils/apollo"

type TUploadResponse = {
  filePath: string
}

export function uploadImage(file: File): Promise<TUploadResponse> {
  const fd = new FormData()
  fd.append('image', file)
  const token = getLocalToken()
  if (!token) {
    throw new Error('no token')
  }
  return fetch(API_HOST + '/api/v1/misc/upload', {
    headers: {
      'Authorization': `Bearer ${token}`
    },
    method: 'POST',
    body: fd,
  }).then(res => res.json())
    .then(response => {
      if ('error' in response) {
        throw new Error(response.error)
      }
      return response
    })
}