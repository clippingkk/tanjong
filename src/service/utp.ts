export enum UTPService {
  clipping = 'clipping',
  book = 'book'
}
export const KonzertThemeMap = {
  classic: {
    id: 0,
    name: 'Classic',
  },
  colorful: {
    id: 1,
    name: 'Colorful',
  },
  dark: {
    id: 2,
    name: 'Noir',
  },
  young: {
    id: 3,
    name: 'Young',
  },
  light: {
    id: 4,
    name: 'Bright',
  }
}

type UTPLinkParams = {
    cid?: number,
    bid: number,
    uid: number,
    theme: number
}

export function getKonzertLink(service: UTPService, params: UTPLinkParams): string {
  const d = new URLSearchParams(params as any)
  const distUrl = `https://konzert.annatarhe.cn/${service.toString()}.html?${d.toString()}`
  return distUrl
}

export function getUTPLink(service: UTPService, params: UTPLinkParams): string {
  const distUrl = encodeURIComponent(getKonzertLink(service, params))
  const screenWidth = 375
  const dpi = 3

  return `https://utp.annatarhe.com/?url=${distUrl}&isMobile=true&isFullPage=true&viewPortWidth=${screenWidth}&width=${screenWidth * dpi}&deviceScaleFactor=${dpi}&viewPortHeight=768`
}