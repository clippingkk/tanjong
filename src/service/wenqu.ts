import { Cache, Key, State } from "swr"
import { WENQU_SIMPLE_TOKEN, WENQU_API_HOST } from "../constants/config"

type WenquErrorResponse = {
  code: number
  error: string
}

export async function wenquRequest<T extends object>(url: string, options: RequestInit = {}): Promise<T> {
  options.headers = {
    ...(options.headers || {}),
    'X-Simple-Check': WENQU_SIMPLE_TOKEN
  }
  options.credentials = 'include'
  options.mode = 'cors'

  try {
    const response: T | WenquErrorResponse = await fetch(WENQU_API_HOST + url, options).then(res => res.json())
    if ('error' in response) {
      throw new Error(response.error)
    }
    return response
  } catch (e) {
    // Sentry.captureException(e)
    return Promise.reject(e)
  }
}

export interface WenquBook {
  id: number
  rating: number
  author: string
  pubdate: string
  totalPages: number
  originTitle: string
  image: string
  doubanId: number
  title: string
  url: string
  press: string
  isbn: string
  tags: string[]
  authorIntro: string
  summary: string
  createdAt: string
  updatedAt: string
}

export interface WenquSearchResponse {
  count: number
  books: WenquBook[]
}

export class WenquSWRCache implements Cache<WenquBook> {
  private _cache = new Map()

  keys(): IterableIterator<string> {
    return this._cache.keys()
  }
  get(key: Key): State<WenquBook, any> | undefined {
    throw new Error("Method not implemented.")
  }
  set(key: Key, value: State<WenquBook, any>): void {
    throw new Error("Method not implemented.")
  }
  delete(key: Key): void {
    throw new Error("Method not implemented.")
  }

}