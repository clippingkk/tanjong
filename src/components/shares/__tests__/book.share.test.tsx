import 'react-native'
import React from 'react'
import { render, screen } from '@testing-library/react-native'
import UTPShareView from '../utp.share'
import { UTPService } from '../../../service/utp'
import { NativeBaseProvider } from 'native-base'
import { NativeSyntheticEvent, NativeScrollEvent, ScrollView } from 'react-native'
import { useScrollHandlers } from 'react-native-actions-sheet'

test('book share should be work', async () => {
  const scrollHandlers = useScrollHandlers<ScrollView>({
    refreshControlBoundary: 0,
  });
  render(
    <NativeBaseProvider initialWindowMetrics={{
      frame: { x: 0, y: 0, width: 0, height: 0 },
      insets: { top: 0, left: 0, right: 0, bottom: 0 },
    }}>
      <UTPShareView
        kind={UTPService.book}
        bookDBID={111111}
        bookID={2222}
        uid={333}
        scrollHandler={scrollHandlers}
      />
    </NativeBaseProvider>
  )
  expect(screen).toMatchSnapshot()
})
