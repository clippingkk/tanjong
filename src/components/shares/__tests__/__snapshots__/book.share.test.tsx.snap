// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`book share should be work 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "backgroundColor": "#f4f4f5",
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        [
          {},
          {
            "dataSet": {},
          },
          {},
        ]
      }
      dataSet={{}}
      style={
        {
          "paddingTop": 16,
        }
      }
    >
      <View>
        <View
          dataSet={{}}
          style={
            {
              "alignItems": "center",
              "display": "flex",
              "justifyContent": "center",
            }
          }
        >
          <View
            dataSet={{}}
            style={
              {
                "paddingBottom": 16,
              }
            }
          >
            <View
              dataSet={{}}
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                colorScheme="primary"
                dataSet={{}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "borderColor": "#d4d4d4",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 10,
                  }
                }
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                  test={true}
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#171717",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      Classic
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 8,
                  }
                }
              />
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                colorScheme="primary"
                dataSet={{}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "borderColor": "#d4d4d4",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 10,
                  }
                }
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                  test={true}
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#171717",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      Colorful
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 8,
                  }
                }
              />
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                colorScheme="primary"
                dataSet={{}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "borderColor": "#d4d4d4",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 10,
                  }
                }
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                  test={true}
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#171717",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      Noir
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 8,
                  }
                }
              />
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                colorScheme="primary"
                dataSet={{}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "borderColor": "#d4d4d4",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 10,
                  }
                }
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                  test={true}
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#171717",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      Young
                    </Text>
                  </View>
                </View>
              </View>
              <View
                dataSet={{}}
                style={
                  {
                    "width": 8,
                  }
                }
              />
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                colorScheme="primary"
                dataSet={{}}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#0891b2",
                    "borderRadius": 4,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingBottom": 10,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 10,
                  }
                }
              >
                <View
                  dataSet={{}}
                  style={
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    }
                  }
                  test={true}
                >
                  <View
                    dataSet={{}}
                    style={{}}
                  >
                    <Text
                      dataSet={{}}
                      style={
                        {
                          "backgroundColor": undefined,
                          "color": "#171717",
                          "fontFamily": undefined,
                          "fontSize": 14,
                          "fontStyle": "normal",
                          "fontWeight": "400",
                          "letterSpacing": 0,
                          "lineHeight": 21,
                          "textDecorationLine": undefined,
                        }
                      }
                    >
                      Bright
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              colorScheme="primary"
              dataSet={{}}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#0891b2",
                  "borderRadius": 4,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 16,
                  "marginTop": 16,
                  "paddingBottom": 10,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 10,
                }
              }
            >
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
                test={true}
              >
                <View
                  dataSet={{}}
                  style={{}}
                >
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    Save Image
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              colorScheme="primary"
              dataSet={{}}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#0891b2",
                  "borderRadius": 4,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 10,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 10,
                }
              }
            >
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
                test={true}
              >
                <View
                  dataSet={{}}
                  style={{}}
                >
                  <Text
                    dataSet={{}}
                    style={
                      {
                        "backgroundColor": undefined,
                        "color": "#171717",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    Share Link
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              [
                {
                  "backgroundColor": "transparent",
                },
                {
                  "aspectRatio": 1.7777777777777777,
                  "borderRadius": 4,
                  "overflow": "hidden",
                  "width": "80%",
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "bottom": 0,
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  {
                    "opacity": 1,
                  },
                ]
              }
            >
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "center",
                    "display": "flex",
                    "justifyContent": "center",
                  }
                }
              >
                <ActivityIndicator />
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaProvider>
`;
